presto-hive/src/main/java/io/prestosql/plugin/hive/s3/S3SecurityMappingConfigurationProvider.java

```diff
@@ -90,7 +90,12 @@ public void updateConfiguration(Configuration configuration, HdfsContext context
 
         S3SecurityMapping mapping = mappings.get().getMapping(context.getIdentity(), uri)
                 .orElseThrow(() -> new AccessDeniedException("No matching S3 security mapping"));
+        if (mapping.isUseClusterDefault()) return;
+        updateConfiguration(configuration, context, mapping);
+    }
 
+    private void updateConfiguration(Configuration configuration, HdfsContext context, S3SecurityMapping mapping)
```

Comment:
no need to extract this function

Reply:
In my opinion such extraction helps to separate logic related to 
- selecting appropriate mapping
- updating mapping configuration itself
During reading the code in case if one is not interested in digging into second option could just skip it.
@kokosing what do you think?

Reply:
This refactor is not related to the actual change and I am not sure its value in general. Let's keep it as it was.

Reply:
Ok.